src/GitHubUtil.js

```diff
@@ -13,16 +13,50 @@ const logApiResult = Log.logApiResult;
 
 const GitHubAuthentication = { type: 'token', username: Config.githubUserLogin(), token: Config.githubToken() };
 
+function commonAppender(allPages, aPage) {
+    allPages.data = allPages.data.concat(aPage.data);
+}
+
+function statusAppender(statusPages, aPage) {
+    let statuses = statusPages.data.statuses;
+    statusPages.data.statuses = statuses.concat(aPage.data.statuses);
+}
+
+function protectedBranchAppender(protectedBranchPages, aPage) {
+    let contexts = protectedBranchPages.data.protection.required_status_checks.contexts;
+    protectedBranchPages.data.protection.required_status_checks.contexts =
+        contexts.concat(aPage.data.protection.required_status_checks.contexts);
+}
+
+async function pager(firstPage, appender) {
+    let allPages = null;
+    if (appender === undefined)
+        appender = commonAppender;
+
+    function doPager (nPage) {
```

Comment:
If you can convert this complex-looking recursion into a simple-looking loop as in the Pagination [example](https://www.npmjs.com/package/@octokit/rest#pagination), please do so.

Reply:
I was thinking about that but, unfortunately, have found nothing simpler. We are facing this  'complication' due to a different paginated item layout, mentioned above.